From: Paul Gofman <pgofman@codeweavers.com>
Subject: [PATCH 2/2] ntdll: Increase kernel stack size.
Message-Id: <20220210140616.488027-2-pgofman@codeweavers.com>
Date: Thu, 10 Feb 2022 17:06:16 +0300
In-Reply-To: <20220210140616.488027-1-pgofman@codeweavers.com>
References: <20220210140616.488027-1-pgofman@codeweavers.com>

Signed-off-by: Paul Gofman <pgofman@codeweavers.com>
---
    The observed problem is seen as semi-random crashes in Resident Evil Village
    reproducible on Nvidia. At least vkCreateComputePipelines() sometimes needs more stack
    space than currently available. Since winevulkan switch to wine_unix_call interface
    most of the native Vulkan calls go from syscall stack which size is quite small. This
    might also be a potential issue with the other native Unix libraries.

 dlls/ntdll/unix/unix_private.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dlls/ntdll/unix/unix_private.h b/dlls/ntdll/unix/unix_private.h
index 11111111111..11111111111 100644
--- a/dlls/ntdll/unix/unix_private.h
+++ b/dlls/ntdll/unix/unix_private.h
@@ -88,7 +88,7 @@ static const SIZE_T page_size = 0x1000;
 static const SIZE_T teb_size = 0x3800;  /* TEB64 + TEB32 + debug info */
 static const SIZE_T signal_stack_mask = 0xffff;
 static const SIZE_T signal_stack_size = 0x10000 - 0x3800;
-static const SIZE_T kernel_stack_size = 0x20000;
+static const SIZE_T kernel_stack_size = 0x60000;
 static const SIZE_T kernel_stack_guard_size = 0x1000;
 static const SIZE_T min_kernel_stack  = 0x3000;
 static const LONG teb_offset = 0x2000;

-- 
2.35.1

